tests/python/unittest/test_tir_ptx_mma.py

```diff
@@ -0,0 +1,1356 @@
+    T.func_attr({"global_symbol": "default_function", "tir.noalias": True})
+    A = T.match_buffer(a, [8, 32], dtype="int4")
+    B = T.match_buffer(b, [8, 32], dtype="uint4")
+    C = T.match_buffer(c, [8, 8], dtype="int32")
+    brow = T.env_thread("blockIdx.y")
+    bcol = T.env_thread("blockIdx.x")
+    tx = T.env_thread("threadIdx.x")
+    T.launch_thread(brow, 1)
+    T.launch_thread(bcol, 1)
+    T.launch_thread(tx, 32)
+    MultiA = T.allocate([8], "int4", scope="local")
+    MultiB = T.allocate([8], "uint4", scope="local")
+    Accum = T.allocate([2], "int32", scope="local")
+    for i in range(2):
+        Accum[i] = T.int32(0)
+
+    for mma_multi_a_col in T.vectorized(8):
+        MultiA[mma_multi_a_col] = A[(tx % 32) // 4, mma_multi_a_col + (tx % 32) % 4 * 8]
+    for mma_multi_b_col in T.vectorized(8):
+        MultiB[mma_multi_b_col] = B[(tx % 32) // 4, mma_multi_b_col + (tx % 32) % 4 * 8]
+    T.evaluate(
+        T.ptx_mma(
+            "m8n8k32",
+            "row",
+            "col",
+            "int4",
+            "uint4",
+            "int32",
+            MultiA,
+            0,
+            MultiB,
+            0,
+            Accum,
+            0,
+            False,
+            dtype="int32",
+        )
+    )
+    for mma_accum_c_id in range(2):
+        C[(tx % 32) // 4, (tx % 32) % 4 * 2 + mma_accum_c_id] = T.load(
+            "int32", Accum, mma_accum_c_id
+        )
+
+
+@tvm.testing.requires_cuda
+def test_gemm_mma_m8n8k32_row_col_s4u4s32():
+    sch = tvm.tir.Schedule(gemm_mma_m8n8k32_row_col_s4u4s32)
+    arch = tvm.contrib.nvcc.get_target_compute_version()
+    major, minor = tvm.contrib.nvcc.parse_compute_version(arch)
+    if major * 10 + minor < 75:
+        # Require at least SM75
+        return
+    cuda_mod = tvm.build(sch.mod, target="cuda")
+
+    ctx = tvm.cuda()
+    A_tvm = tvm.nd.empty([8, 32], "int4", ctx)
+    B_tvm = tvm.nd.empty([8, 32], "uint4", ctx)
+    C_tvm = tvm.nd.empty([8, 8], "int32", ctx)
+
+    cuda_mod(A_tvm, B_tvm, C_tvm)
+    # Currently the correctness is not checked.
+    # TODO: add correctness checking here.
+
+
+@T.prim_func
+def gemm_mma_m16n8k8_row_col_fp16fp16fp32(a: T.handle, b: T.handle, c: T.handle):
+    T.func_attr({"global_symbol": "default_function", "tir.noalias": True})
+    A = T.match_buffer(a, [16, 8], dtype="float16")
+    B = T.match_buffer(b, [8, 8], dtype="float16")
+    C = T.match_buffer(c, [16, 8], dtype="float32")
+    brow = T.env_thread("blockIdx.y")
+    bcol = T.env_thread("blockIdx.x")
+    tx = T.env_thread("threadIdx.x")
+    T.launch_thread(brow, 1)
+    T.launch_thread(bcol, 1)
+    T.launch_thread(tx, 32)
+    MultiA = T.allocate([4], "float16", scope="local")
+    MultiB = T.allocate([2], "float16", scope="local")
+    Accum = T.allocate([4], "float32", scope="local")
+    for i in range(4):
+        Accum[i] = T.float32(0)
+
+    for mma_multi_a_col in T.vectorized(4):
```

Comment:
Thanks for the PR! I wonder if you could elaborate more on the necessity of the declarations of `MultiA`, `MultiB` and `Accum` buffers here. Do buffers like `A`, `B` and `C` not work within the MMA assembly code generated below?

Reply:
To use MMA instructions, the multiplicands and accumulator should be placed in registers, otherwise, the behavior is undefined. I have tried to use global buffers (e.g., `A`, `B`, `C`) to invoke MMA instructions, and the results are all wrong.